### DISCUSSION TITLE
Terminology

### DISCUSSION THREAD

**Discussion by id-entity** [geometor/relop#4] [General]
I suggest we call < > "relational operators" instead of "brackets". The working title of the foundational theory is Relop, from relational operators, and the philosophical approach is foundationally relational, not data-object oriented. Data structures should be tools, not foundational concepts in this approach. 

For general data structures I suggest we follow/apply Norman's approach as suitable: 
https://www.youtube.com/watch?v=q2beQrKjtzs

**Comment by phiarchitect** [comment]
I like `relop`
changed the repo to 'relop'
description to 'foundational study of relational operators < > '
and updated readme

**Comment by phiarchitect** [comment]
defining the data structures and representing them symbolically will be very important

**Comment by id-entity** [comment]
As said, Norman's 2by2 matrix of repeatability and order seems very helpful and interesting at least as the starting area. But as [ ] and [] are very cruxial primitives of concatenation, seems we might need something more generally aplicable than Norman's notation. So far no concrete suggestion for such.

**Comment by id-entity** [comment]
Characters < > in a Relop word can be defined as a list. The characters repeat and are ordered by some generative algorithm into linear strings.

The word lists are anti-concatenated into words by whitespace aka [ ], and concatenation of two words into a new word is the operation [ ] -> []. 
  
Reading and interpreting a word-list numerically involves at least following steps in the example-process of proceeding from <>< to 1:1.   
1) Interpreting a <> and multiple strings of < or > as a syllabus. E.g. a word <><<><< consists of syllabi <>< and <><<. 
2) In inverse, a syllabus consists of a < or > and multiple strings of <>. For example <>><><>> consists of syllabi <>> and <><>>. 
3) If we compare this with <> as "consonant" and < or > as vowels, First type could be thus informally called "vowell heavy" and second "consonant heavy".  
4) With syllabus structure of "consonants" and "vowells" we can proceed to reading a relop word- as an unordered and repeating mset 
5) and count the number of consonants and vowels in a relop word.
6) The counting produces number pairs which have various interpretations, such as
7) unordered and non-repeating 2-sets {2 3} which have further interpretation as
8) ordered and non-repeating oset of a generated "SB row"*. 
9) Here we can observe that the process has involved all four data types from a) list to b) mset to c) set and d) oset. 

*The generation of Stern-Brocot type (SB) structures, ie. words and rows which allow SB has not been included in the description of the process, only the interpretative procedure.

**Comment by phiarchitect** [comment]
this last comment is just the breakdown I've been looking for

**Comment by phiarchitect** [comment]
I need to get clear on the following:

- consonants
- vowels
- word
- syllabus
- row

And then I need to understand how the operation is applied

**Comment by id-entity** [comment]
### Generating Function

'Word' and 'Row' are primarily defined for the generative algorithm of printing expressions

Word separator: [ ]
Concatenation: [ ] -> []
word: [ ]word[ ] (so far a word can consist of only charactes < and/or >)
Concatenation of 2 words into 1: 
[ ]Lword[ ]Rword[ ]
[ ]Lword[]Rword[ ]
The [] between concatenated word is here redundant and does not add any clarity, so lets drop it for now: 
[ ]LwordRword[ ]
(on the other hand, it might be very useful and important data to keep and express especially the Fibonacci word concatenation history... more to chew, hmm)

Row is a combination of at least 2 words of a seed Row. For the purposes of SB investigation, all rows are palindromes. So far I've used ENTER as the row separator in my word processor. E.g. Next/New Row "NR" could function as Row separator in the generated data files. 

A New Row is formed by filling all the word separating mediant cells [ ] of the previous row (e.g. seed row) with concatenated mediants from the L and R words of a [ ]. 

NRS<[ ]> (2 words, 1 word separator aka mediant cell where new concatenated LRword is inserted)
NR1<[ ]<>[ ]> (3 words, 2 cells for inserting new words)
NR2<[ ]<<>[ ]<>[ ]<>>[ ]> (5 words, 4 cells)

More generally, L and R as variables for words. 

NRS L[ ]R 
NR1 L[ ]LR[ ]R 
NR2 L[ ]LLR[ ]LR[ ]LRR[ ]R 


### Interpretative Function

Vowels, consonants and syllabi are defined only for interpretative algorithm of expressions

Voweltype1: <
Voweltype2: >
Consonant: <>
Syllabus: Ordered concatenation of 1 or more consonants (c) and 1 or more vowels (v), either  c(n)v(n) or v(n)c(n), n for number of vowels and consonants in a syllabus. 

Note: a word does not have to be a syllabus, it can consist of only vowels or consonants. Word is defined for generating data. Syllabi, vowels and consonant for reading and interpreting data numerically. Syllabus-concept seems so far necessary mainly or only for defining "_interesting_ words" and classifying the generator seeds.

**Comment by phiarchitect** [comment]
this last post is a tremendous help!
